scripts/setDefaultIndex.py

```diff
@@ -0,0 +1,145 @@
+
+def search_index_and_type(es_index, es_type, query):
+    search_response_raw = json.dumps(es.search(index=es_index,
+                                               doc_type=es_type,
+                                               q=query,
+                                               request_timeout=10),
+                                     indent=indent_level)
+    search_response_json = json.loads(search_response_raw)
+    hits_json = safe_list_read(search_response_json, 'hits')
+    search_number_of_hits = safe_list_read(hits_json, 'total')
+    return search_number_of_hits
+
+def verify_document_for_content(es_index, es_type, content):
+    missing_elements = {}
+    content_json = json.loads(json.dumps(content))
+    for key in content_json.keys():
+        query = key + ':' + '"'+ content_json[key] + '"'
+        hits = search_index_and_type(es_index, es_type, query)
+        if hits > 0:
+            logging.debug(str(hits) + " hit(s) for " + str(query) + ".")
+        else:
+            logging.debug("No hits for " + str(query) + ". Attempting to update index...")
+            missing_elements[key] = content_json[key]
+    return missing_elements
+
+def create_document(es_index, es_type, es_id, es_body):
+    return json.dumps(es.create(index=es_index,
+                                doc_type=es_type,
+                                id=es_id,
+                                body=es_body,
+                                request_timeout=10),
+                      indent=indent_level)
+
+def create_document_if_it_doesnt_exist(es_index, es_type, es_id, es_body):
+    document_exists = es.exists(index=es_index, doc_type=es_type, id=es_id, request_timeout=10)
+    if not document_exists:
+        logging.info('Document %s/%s/%s/%s does not exist. Creating it now...', localhost, es_index, es_type, es_id)
+        document_created = create_document(es_index, es_type, es_id, es_body)
+        logging.info("Create document returns: \n %s", document_created)
+        return created
+    else:
+        logging.info("Document %s/%s/%s/%s already exists.", localhost, es_index, es_type, es_id)
+        return not created
+
+
+def sleep_if_necessary(need_to_sleep):
+    if need_to_sleep:
+        time.sleep(1)
+
+# ----------------- MAIN -----------------
+def main():
+    logging.debug("================================== INDEX PATTERN ==================================")
+    index_pattern_doc_created = create_document_if_it_doesnt_exist(kibana_index,
+                                                                   index_pattern_type,
+                                                                   nm_index_pattern,
+                                                                   index_pattern_content)
+    sleep_if_necessary(index_pattern_doc_created) # ES doesn't show the index-pattern updates immediately after inserting them
```

Comment:
is this guaranteed to be available after the 1 second sleep? if not we might want to adjust it to do an active polling or something